File: 3751-total-waviness-of-numbers-in-range-i/3751-total-waviness-of-numbers-in-range-i.py
class Solution:
    def totalWaviness(self, num1: int, num2: int) -> int:
        res = 0
        for i in range(num1, num2 + 1):
            strInt = str(i)
            for j in range(1, len(strInt) - 1):
                if (int(strInt[j]) > int(strInt[j - 1]) and int(strInt[j]) > int(strInt[j + 1])) or (int(strInt[j]) < int(strInt[j - 1]) and int(strInt[j]) < int(strInt[j + 1])):
                    res += 1
        return res

File: 3751-total-waviness-of-numbers-in-range-i/3751-total-waviness-of-numbers-in-range-i.cpp
class Solution {
public:
    int totalWaviness(int num1, int num2) {
        int res = 0;
        for(int i = num1; i <= num2; i++)
        {
            string strInt = to_string(i);
            for(int j = 1; j < strInt.length() - 1; j++)
            {
                if((strInt[j] > strInt[j - 1] && strInt[j] > strInt[j + 1]) || (strInt[j] < strInt[j - 1] && strInt[j] < strInt[j + 1]))
                {
                    res++;
                }
            }
        }
        return res;
    }
};